coordinator/src/node/connection.rs

```diff
@@ -0,0 +1,100 @@
+use lightning::ln::msgs::NetAddress;
+use ln_dlc_node::node::Node;
+use ln_dlc_node::node::NodeInfo;
+use ln_dlc_node::node::PaymentMap;
+use rand::seq::SliceRandom;
+use rand::thread_rng;
+use std::net::IpAddr;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use std::time::Duration;
+
+const CONNECTION_CHECK_INTERVAL: Duration = Duration::from_secs(30);
+
+pub async fn keep_public_channel_peers_connected(node: Arc<Node<PaymentMap>>) {
+    loop {
+        reconnect_to_disconnected_public_channel_peers(node.clone());
+
+        tokio::time::sleep(CONNECTION_CHECK_INTERVAL).await;
+    }
+}
+
+fn reconnect_to_disconnected_public_channel_peers(node: Arc<Node<PaymentMap>>) {
```

Comment:
🙃 nit: it would have been more readable if we only passed in what we actually needed here (peer_manager, channel_manager, network_graph) rather than the whole lot

Reply:
That's what I started with but `connect` is implemented on `Node` and I didn't feel like refactoring that.